snuba/query/conditions.py

```diff
@@ -321,3 +321,23 @@ def build_match(
             ),
         ]
     )
+
+
+def get_literals_for_column_condition(
+    col: str, exp_type: Any, conditions: Expression
+) -> Sequence[Literal]:
+    match = build_match(col, (ConditionFunctions.EQ,), exp_type)
+    for exp in conditions:
+        result = match.match(exp)
+        if result is not None:
+            rhs = result.expression("rhs")
+            if isinstance(rhs, Literal):
+                return [rhs]
+            elif isinstance(rhs, FunctionCall):
+                params = []
+                for param in rhs.parameters:
+                    assert isinstance(param, Literal)
+                    params.append(param)
```

Comment:
This branch seems kind of unintuitive to me. I think I get why you did it (because project ids happen to be typically in an array) but it doesn't seem particularly generalisable.

Reply:
Even if this only ever works for org_id and project_id this will be useful. Now that project_ids are not in a separate extension, the only way to get the project_ids in a query is to do something like this. I've rewritten this code at least twice before this.

Reply:
Yeah that makes sense. I was just worried we'd return false positives here. Like is there a reason we can't check function name against an allowed list, instead of just matching any function at all that's called with an int?

Reply:
`build_match` is already doing that. If the match above has succeeded, then I already know the function is `array` or `tuple`.

Reply:
Cool, didn't see that. Thanks.